src/rules.jl

```diff
@@ -101,10 +107,18 @@ julia> hypot_pullback(1) == (NoTangent(), (x / hypot(x, y)), (y / hypot(x, y)))
 true
 ```
 
-See also: [`frule`](@ref), [`@scalar_rule`](@ref)
+The optional [`RuleConfig`](@ref) option allows specifying rrules only for AD systems that
+support given features. If not needed, then it can be omitted and the `rrule` without it
+will be hit as a fallback. This is the case for most rules.
+
+See also: [`frule`](@ref), [`@scalar_rule`](@ref), [`RuleConfig`](@ref)
 """
 rrule(::Any, ::Vararg{Any}) = nothing
 
+# if no config is present then fallback to config-less rules
+rrule(::RuleConfig, f, args...; kwargs...) = rrule(f, args...; kwargs...)
+#TODO do we need to do something for kwargs special here for performance?
```

Comment:
should we open an issue about this instead?

Reply:
Good idea.
```suggestion
# TODO do we need to do something for kwargs special here for performance?
# See: https://github.com/JuliaDiff/ChainRulesCore.jl/issues/368
```